### PR TITLE
Node properties

### PR DESCRIPTION
**Pull request by OskarDamkjaer**
image

### CODE REVIEW

File: src/browser/modules/D3Visualization/components/DetailsPane.tsx [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+  if (!properties.length) {
+    return <div>No properties to display</div>
+  }
+
+  const allItemProperties = [
+    { key: '<id>', value: `${id}` },
+    ...properties
+  ].sort((a, b) => (a.key < b.key ? -1 : 1))
+  const formatForDisplay = (neo4jValue: unknown) =>
+    unescapeDoubleQuotesForDisplay(
```
Comment (jharris4) [inline src/browser/modules/D3Visualization/components/DetailsPane.tsx:58]: are these needed? remove if not

File: src/browser/modules/D3Visualization/components/DetailsPane.tsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+      stringifyMod(neo4jValue, stringModifier, true)
+    )
+
+  return (
+    <>
+      <StyledInspectorClipboardCopyAll>
+        <div style={{ marginLeft: 'auto', marginRight: '5px' }}>
```
Comment (jharris4) [inline src/browser/modules/D3Visualization/components/DetailsPane.tsx:65]: inline styles... move to styled.tsx...

File: src/browser/modules/D3Visualization/components/DetailsPane.tsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+                  id={vizItem.item.id}
+                  properties={vizItem.item.properties}
+                />
+              </StyledInlineList>
+            )}
+
+            {vizItem.type === 'relationship' && (
+              <StyledInlineList>
+                <StyledTokenRelationshipType
+                  key="token"
```
Comment (jharris4) [inline src/browser/modules/D3Visualization/components/DetailsPane.tsx:146]: key can be removed here?

File: src/browser/modules/D3Visualization/components/Explorer.tsx [hunk 4]
```diff
@@ -121,13 +135,18 @@ export class ExplorerComponent extends Component<any, ExplorerComponentState> {
   }
 
   onGraphModelChange(stats: any) {
-    this.setState({ stats: stats })
+    this.setState((oldState: any) => ({
```
Comment (jharris4) [inline src/browser/modules/D3Visualization/components/Explorer.tsx:138]: could this be made simpler? Also, are we sure we won't be keeping around old/stale stats by spreading the old ones like this?

File: src/browser/modules/D3Visualization/components/Explorer.tsx [hunk 5]
```diff
@@ -139,7 +158,7 @@ export class ExplorerComponent extends Component<any, ExplorerComponentState> {
 
   onSelectedRelType(relType: any, propertyKeys: any) {
     this.setState({
-      selectedItem: {
+      selectedLabel: {
```
Comment (jharris4) [inline src/browser/modules/D3Visualization/components/Explorer.tsx:161]: shouldn't this be `selectedType` ?

File: src/browser/modules/D3Visualization/components/Explorer.tsx [hunk 6]
```diff
@@ -174,73 +193,44 @@ export class ExplorerComponent extends Component<any, ExplorerComponentState> {
...
+        <StyledGraphAreaContainer>
+          <GraphComponent
+            fullscreen={this.props.fullscreen}
+            frameHeight={this.props.frameHeight}
+            relationships={this.state.relationships}
+            nodes={this.state.nodes}
+            getNodeNeighbours={this.getNodeNeighbours.bind(this)}
+            onItemMouseOver={this.onItemMouseOver.bind(this)}
+            onItemSelect={this.onItemSelect.bind(this)}
+            graphStyle={this.state.graphStyle}
```
Comment (jharris4) [inline src/browser/modules/D3Visualization/components/Explorer.tsx:215]: should this not be `graphStyle` using the above variable like in the props for `<NodeInspectorPanel>` ?

File: src/browser/modules/D3Visualization/components/GrassEditor.tsx [hunk 7]
```diff
@@ -300,10 +301,13 @@ export class GrassEditorComponent extends Component<any> {
       return null
     }
     return (
-      <StyledInlineList className="style-picker">
+      <StyledInlineListStylePicker
+        className="style-picker"
```
Comment (jharris4) [inline src/browser/modules/D3Visualization/components/GrassEditor.tsx:305]: why the className here?

File: src/browser/modules/D3Visualization/components/ResultsPane.tsx [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+  selectedLabel,
+  stats,
+  hasTruncatedFields
+}: ResultPaneProps): JSX.Element {
+  const { relTypes, labels, nodeCount, relationshipCount } = stats
+
+  return (
+    <StyledLegend>
+      <div
+        style={{
```
Comment (jharris4) [inline src/browser/modules/D3Visualization/components/ResultsPane.tsx:71]: styled.tsx for this?

File: src/browser/modules/D3Visualization/components/styled.tsx [hunk 9]
```diff
@@ -146,22 +152,41 @@ export const StyledInspectorFooter = styled.div`
...
+
 export const StyledInspectorFooterRowListKey = styled.div`
   float: left;
   font-weight: 800;
+  flex: 1;
 `
 
 export const StyledInspectorFooterRowListValue = styled.div`
-  padding-left: 3px;
+  padding: 0 3px;
```
Comment (jharris4) [inline src/browser/modules/D3Visualization/components/styled.tsx:185]: do we want the right padding here?

File: src/browser/modules/D3Visualization/components/styled.tsx [hunk 10]
```diff
@@ -170,6 +195,16 @@ export const StyledInlineList = styled.ul`
   word-break: break-word;
 `
 
+export const StyledInlineListStylePicker = styled(StyledInlineList)<{
+  frameHeight: number
+}>`
+  display: grid;
+  overflow-y: auto;
+  max-height: ${props => props.frameHeight - 75}px;
```
Comment (jharris4) [inline src/browser/modules/D3Visualization/components/styled.tsx:203]: Is this 75 the header height? If so is there a constant we can import for it, or if not, create one?

File: src/browser/modules/D3Visualization/components/styled.tsx [hunk 11]
```diff
@@ -209,17 +252,43 @@ export const StyledStatus = styled.div`
   overflow: auto;
 `
 
+export const StyledDetailsStatus = styled.div`
+  position: relative;
+  float: left;
+  padding-left: 16px;
+  margin-top: 0;
```
Comment (jharris4) [inline src/browser/modules/D3Visualization/components/styled.tsx:259]: duplicate style to be removed

File: src/browser/modules/D3Visualization/components/styled.tsx [hunk 12]
```diff
@@ -265,12 +336,11 @@ export const StyledLegendRow = styled.div`
 `
 
 export const StyledLegend = styled.div`
-  background-color: ${props => props.theme.frameBackground};
   position: absolute;
   z-index: 1;
-  top: 0;
   right: 0;
```
Comment (jharris4) [inline src/browser/modules/D3Visualization/components/styled.tsx:341]: having both left & right specified is a bad code smell imo

File: src/browser/modules/D3Visualization/components/GrassEditor.tsx (hunk 7 above)
Comment (OskarDamkjaer) [reply src/browser/modules/D3Visualization/components/GrassEditor.tsx]: Removed! Old and unused

File: src/browser/modules/D3Visualization/components/Explorer.tsx (hunk 4 above)
Comment (OskarDamkjaer) [reply src/browser/modules/D3Visualization/components/Explorer.tsx]: Fixed

File: src/browser/modules/D3Visualization/components/Explorer.tsx (hunk 5 above)
Comment (OskarDamkjaer) [reply src/browser/modules/D3Visualization/components/Explorer.tsx:207]: Probably. Not looked into this yet

File: src/browser/modules/D3Visualization/components/styled.tsx (hunk 10 above)
Comment (OskarDamkjaer) [reply src/browser/modules/D3Visualization/components/styled.tsx:136]: There's definitely a need for more constants, haven't got a round to it yet. Looking to do so tomorrow

File: src/browser/modules/D3Visualization/components/Explorer.tsx (hunk 5 above)
Comment (OskarDamkjaer) [reply src/browser/modules/D3Visualization/components/Explorer.tsx:207]: It should be `selectedItem`!

Comment (OskarDamkjaer) [reply src/browser/modules/D3Visualization/components/Explorer.tsx:207]: But fixed now

File: src/browser/modules/D3Visualization/components/DetailsPane.tsx [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+  const allItemProperties = [
+    { key: '<id>', value: `${vizItem.item.id}`, type: 'string' },
+    ...vizItem.item.properties
+  ].sort((a, b) => (a.key < b.key ? -1 : 1))
+
+  return (
+    <>
+      <PaneHeader>
+        {vizItem.type === 'relationship' && (
+          <StyledTokenRelationshipType
```
Comment (jharris4) [inline src/browser/modules/D3Visualization/components/DetailsPane.tsx:60]: popup missing here, and grass editor?

File: src/browser/modules/D3Visualization/components/Explorer.tsx [hunk 14]
```diff
@@ -174,73 +227,48 @@ export class ExplorerComponent extends Component<any, ExplorerComponentState> {
...
-        <InspectorComponent
-          hasTruncatedFields={this.props.hasTruncatedFields}
-          fullscreen={this.props.fullscreen}
-          hoveredItem={this.state.hoveredItem}
-          selectedItem={this.state.selectedItem}
-          graphStyle={this.state.graphStyle}
-          onExpandToggled={this.onInspectorExpandToggled.bind(this)}
-        />
+      <StyledFullSizeContainer id="svg-vis">
+        <StyledGraphAreaContainer>
```
Comment (jharris4) [inline src/browser/modules/D3Visualization/components/Explorer.tsx:240]: Is the extra component needed here?

**Review by jharris4 — APPROVED** [review]
Looking good from a code/functionality perspective! I left two minor comments for you to address if you'd like, but I'm approving now.

File: src/browser/modules/D3Visualization/components/DetailsPane.tsx (hunk 13 above)
Comment (OskarDamkjaer) [reply src/browser/modules/D3Visualization/components/DetailsPane.tsx:60]: Handling this in a followup pr

File: src/browser/modules/D3Visualization/components/Explorer.tsx (hunk 14 above)
Comment (OskarDamkjaer) [reply src/browser/modules/D3Visualization/components/Explorer.tsx:240]: Handling this in a followup pr

**Pull request merged**